src/lib.rs

```diff
@@ -310,6 +339,110 @@ pub trait PrimeField: Field + From<u64> {
     /// It can be calculated by exponentiating `Self::MULTIPLICATIVE_GENERATOR` by `t`,
     /// where `t = (modulus - 1) >> Self::S`.
     const ROOT_OF_UNITY: Self;
+
+    /// Inverse of [`Self::ROOT_OF_UNITY`].
+    const ROOT_OF_UNITY_INV: Self;
+
+    /// Generator of the `t-order` multiplicative subgroup.
+    ///
+    /// It can be calculated by exponentiating [`Self::MULTIPLICATIVE_GENERATOR`] by `2^s`,
+    /// where `s` is [`Self::S`].
+    const DELTA: Self;
+}
+
+/// The subset of prime-order fields such that `(modulus - 1)` is divisible by `N`.
+///
+/// In these fields, there will be two valid choices of [`Self::ZETA`]. Similarly to
+/// [`PrimeField::MULTIPLICATIVE_GENERATOR`], the specific choice does not matter, as long
+/// as the choice is consistent across all uses of the field.
```

Comment:
I think that "two valid choices" only applies when $N = 3$. In other cases, there will be $N-1$ choices for prime $N$.

Our curve generation scripts have always chosen $\zeta$ for $N = 3$ to be the smaller of the two possibilities. I recommend this for interoperability.

```suggestion
/// If `N` is prime, there will be `N - 1` valid choices of [`Self::ZETA`]. Similarly to
/// [`PrimeField::MULTIPLICATIVE_GENERATOR`], the specific choice does not matter, as long
/// as the choice is consistent across all uses of the field. Choosing the value that is
/// smallest, when considered as an integer, may help to ensure consistency.
```

Reply:
Good catch. The "two valid choices" language was from then this trait only represented N = 3 that FieldExt did, before I suggested we generalize it.